librb/include/rb_memory.h

```diff
@@ -37,6 +37,7 @@
 
 void rb_outofmemory(void) __attribute__((noreturn));
 
+__attribute__((malloc, alloc_size(1), returns_nonnull))
```

Comment:
Please use macros for this, so that we can provide it on other compilers in the future.